packages/target-electron/src/deltachat/webxdc.ts

```diff
@@ -826,50 +861,118 @@ export default class DCWebxdc {
   }
 }
 
-async function webxdcProtocolHandler(
+/**
+ * Make sure to only `return makeResponse()` because it sets headers
+ * that are important for security, namely `Content-Security-Policy`.
+ * Failing to set CSP might result in the app being able to create
+ * an <iframe> with no CSP, e.g. `<iframe src="/no_such_file.lol">`
+ * within which they can then do whatever
+ * through the parent frame, see
+ * "XDC-01-002 WP1: Full CSP bypass via desktop app webxdc.js"
+ * https://public.opentech.fund/documents/XDC-01-report_2_1.pdf
+ */
+const makeResponse = (
+  appId: string | null,
```

Comment:
Is there not a less confusing way to specify if we want lax CSP? Why is `appId` sometimes provided and sometimes not? I now see why, but it's not immediately clear. I suggest to instead add an `options` object with an optional `cspAllowMapsImgSrc`.